src/pkg/scan/export/filter_processor.go

```diff
@@ -0,0 +1,142 @@
+	projects, err := dfp.projectMgr.List(ctx, query)
+
+	if err != nil {
+		return nil, err
+	}
+	logger.Infof("Selected %d projects administered by user %s ", len(projects), userName)
+	projectIds := make([]int64, 0)
+	for _, proj := range projects {
+		projectIds = append(projectIds, proj.ProjectID)
+	}
+
+	// check if the project ids specified in the filter are present in the list
+	// of projects of which the current user is a project admin
+	if len(projectIdsToFilter) == 0 {
+		return projectIds, nil
+	}
+	m := make(map[int64]bool)
+	for _, projectID := range projectIds {
+		m[projectID] = true
+	}
+	filtered := make([]int64, 0)
+
+	for _, filteredProjID := range projectIdsToFilter {
+		if m[filteredProjID] {
+			filtered = append(filtered, filteredProjID)
+		}
+	}
+	return filtered, nil
+}
+
+func (dfp *DefaultFilterProcessor) ProcessRepositoryFilter(ctx context.Context, filter string, projectIds []int64) ([]*selector.Candidate, error) {
+	sel := doublestar.New(doublestar.RepoMatches, filter, "")
+	candidates := make([]*selector.Candidate, 0)
+
+	for _, projectID := range projectIds {
+		query := q.New(q.KeyWords{"ProjectID": projectID})
+		allRepos, err := dfp.repoMgr.List(ctx, query)
+		if err != nil {
+			return nil, err
+		}
+		for _, repository := range allRepos {
+			namespace, repo := utils.ParseRepository(repository.Name)
+			candidates = append(candidates, &selector.Candidate{NamespaceID: repository.RepositoryID, Namespace: namespace, Repository: repo, Kind: "image"})
+		}
+	}
+	// no repo filter specified then return all repos across all projects
+	if filter == "" {
+		return candidates, nil
+	}
+	return sel.Select(candidates)
+}
+
+func (dfp *DefaultFilterProcessor) ProcessTagFilter(ctx context.Context, filter string, repositoryIds []int64) ([]*selector.Candidate, error) {
+	sel := doublestar.New(doublestar.Matches, filter, "")
+	candidates := make([]*selector.Candidate, 0)
+
+	for _, repoID := range repositoryIds {
+		query := q.New(q.KeyWords{"RepositoryID": repoID})
+		allTags, err := dfp.tagMgr.List(ctx, query)
+		if err != nil {
+			return nil, err
+		}
+		cand := &selector.Candidate{NamespaceID: repoID, Kind: "image"}
+		for _, tag := range allTags {
+			cand.Tags = append(cand.Tags, tag.Name)
+		}
+		candidates = append(candidates, cand)
+	}
+	// no tags specified then simply return all the candidates
+	if filter == "" {
+		return candidates, nil
+	}
+	return sel.Select(candidates)
+}
+
+func (dfp *DefaultFilterProcessor) getProjectQueryFilter(user *commonmodels.User) *q.Query {
+	secContext := local.NewSecurityContext(user)
+	if secContext.IsSysAdmin() {
+		logger.Infof("User %v is sys admin. Selecting all projects for export.", user.Username)
+		return q.New(q.KeyWords{})
+	}
+	logger.Infof("User %v is not sys admin. Selecting projects with admin roles for export.", user.Username)
+	return q.New(q.KeyWords{"member": &models.MemberQuery{UserID: user.UserID, Role: common.RoleProjectAdmin}})
```

Comment:
This behavior means if the user is not system admin, he can only list projects for which he is the project administrator? What happened If the user is the Developer role?

Reply:
Yes currently that is the implementation. I am facing a technical limitation in using the `user.MemberQuery` struct - it only allows one role at a time. I would not be in favor of writing another raw SQL for getting users with specific set of roles. Is there a role I can use which is a combination of both `Project Admin` and `Project Developer`?
cc @wy65701436

Reply:
let's merge this PR firstly, then use other PR to fix.